context-free.md

```diff
@@ -1,21 +1,32 @@
-# Context Free
+# Context Free Questions
+#### Zeng Wei Cheng
 
 ##  Who is this programming language for?
+Context-free is a external domain specific language designed specifically for artists who have little knowledge in programming. Users of this language can easily get familiar with its syntax or symbols and use it to create art work.
 
 
 ## What is easy to do in this language? Why is it easy?
+It is quite easy to generate complex images with just a few lines of code. It creates these complex images by flipping, rotating, skewing, basic shapes including triangles, circles and squares, all of which can be done by a single statement. Moreover, context-free is easy to generate variations; that is, the same piece of code can generate totally different images. And the user of this program can trace back a specific variation by identifying the three-letter-code shown.
+It is easy because context-free is a good DSL design, allowing readers to easily understand its meaning with meaning variable names. For example, context-free uses jargons such as skew, hue, and saturate. . .etc. And its syntax is easy too. It uses space to separate and it allows users to switch the places of parameters.
```

Comment:
I completely agree with the point you've made about it being easy to generate complex patterns using very few lines of codes and that this is made easier by the functions and shapes used by ContextFree. This is something which is also apparent in your own design - which is incredible. I find it incredibly fascinating that your image is something so colourful and abstract-seeming in spite of being generated basis a set of fixed rules - and I think that this actually is in accordance with your first statement. Additionally, I also completely agree with your second statement about almost the same code being able to generate totally different images; in fact, to verify it - I tried experimenting with your code and by only changing 2 words (SQUARE to TRIANGLE in the 11th line and 120 to 10 in the 4th line), a completely different pattern was created. Yes, I also found that ContextFree allows readers to easily understand the different methods' functionality by using meaningful names.
<img width="1042" alt="screen shot 2018-03-15 at 4 50 33 pm" src="https://user-images.githubusercontent.com/24552462/37461688-35e6910c-2875-11e8-8265-722137090b00.png">
<img width="391" alt="screen shot 2018-03-15 at 4 51 31 pm" src="https://user-images.githubusercontent.com/24552462/37461689-362d728e-2875-11e8-9997-4afbaaf95966.png">